hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/AccountStore.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.token.impl;
+
+import static com.hedera.node.app.spi.state.StateKey.ACCOUNT_STORE;
+import static com.hedera.node.app.spi.state.StateKey.ALIASES_STORE;
+import static com.hedera.services.utils.EntityIdUtils.isAlias;
+
+import com.google.protobuf.ByteString;
+import com.hedera.node.app.spi.meta.TransactionMetadata;
+import com.hedera.node.app.spi.meta.impl.SigTransactionMetadata;
+import com.hedera.node.app.spi.state.State;
+import com.hedera.node.app.spi.state.States;
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hedera.services.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.Transaction;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+
+/**
+ * Provides methods for interacting with the underlying data storage mechanisms for working with
+ * Accounts.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail.
+ */
+public class AccountStore {
```

Comment:
Maybe it is non-final for mocking in unit tests? That would make sense. Or have an interface and a concrete class, so tests can be written to the interface instead of the concrete impl.